addon/utils/bind-data-test-attributes.js

```diff
@@ -0,0 +1,13 @@
+const TEST_SELECTOR_PREFIX = /data-test-.*/;
+
+export default function bindDataTestAttributes(component) {
+  let attributeBindings = component.getWithDefault('attributeBindings', []);
+
+  for (let attr in component) {
+    if (TEST_SELECTOR_PREFIX.test(attr)) {
+      attributeBindings.push(attr);
+    }
+  }
+
+  component.set('attributeBindings', attributeBindings);
```

Comment:
I'm now checking if `attributeBindings` is a computed property in the `bindDataTestAttributes()` function and if so, I skip the automatic binding and display a warning message.